lib/transaction.py

```diff
@@ -933,18 +959,18 @@ def sign(self, keypairs, *, use_cache=False):
                     continue
                 print_error(f"adding signature for input#{i} sig#{j}; {kname}: {_pubkey} schnorr: {self._sign_schnorr}")
                 sec, compressed = keypairs.get(_pubkey)
-                self._sign_txin(i, j, sec, compressed, use_cache=use_cache)
+                self._sign_txin(i, j, sec, compressed, use_cache=use_cache, ndata=ndata)
         print_error("is_complete", self.is_complete())
         self.raw = self.serialize()
 
-    def _sign_txin(self, i, j, sec, compressed, *, use_cache=False):
+    def _sign_txin(self, i, j, sec, compressed, *, use_cache=False, ndata=None):
         '''Note: precondition is self._inputs is valid (ie: tx is already deserialized)'''
         pubkey = public_key_from_private_key(sec, compressed)
         # add signature
         nHashType = 0x00000041 # hardcoded, perhaps should be taken from unsigned input dict
         pre_hash = Hash(bfh(self.serialize_preimage(i, nHashType, use_cache=use_cache)))
         if self._sign_schnorr:
-            sig = self._schnorr_sign(pubkey, sec, pre_hash)
+            sig = self._schnorr_sign(pubkey, sec, pre_hash, ndata)
```

Comment:
If we went with the kw-arg-only approach for `_schnorr_sign` this would be the place to make it say `ndata=ndata`.